rules/macos/defense_evasion_modify_environment_launchctl.toml

```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2021/01/14"
+maturity = "production"
+updated_date = "2021/01/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies modification of an environment variable using launchctl built-in command. Adversaries may execute their own
+malicious payloads by hijacking certain environment variables to load arbitrary libraries or bypass certain
+restrictions.
+"""
+from = "now-9m"
+index = ["auditbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Environment Variable Modification using Launchctl"
+references = [
+    "https://github.com/rapid7/metasploit-framework/blob/master//modules/post/osx/escalate/tccbypass.rb",
+]
+risk_score = 47
+rule_id = "7453e19e-3dbf-4e4e-9ae0-33d6c6ed15e1"
+severity = "medium"
+tags = ["Elastic", "Host", "macOS", "Threat Detection", "Defense Evasion"]
+type = "query"
+
+query = '''
+event.category:process and event.type:(start or process_started) and
+  process.name:launchctl and
+  process.args:(setenv and not (JAVA*_HOME or RUNTIME_JAVA_HOME or DBUS_LAUNCHD_SESSION_BUS_SOCKET or ANT_HOME))
```

Comment:
could benefit from case insensitivity but will leave up to you

Reply:
not necessary: most process args are sensitive and cmd will fail if insensitive.